minicbor-derive/src/encode.rs

```diff
@@ -115,6 +118,19 @@ fn on_enum(inp: &mut syn::DeriveInput) -> syn::Result<proc_macro2::TokenStream>
                         Ok(())
                     }
                 },
+                Encoding::Array if flat & attrs.tag().is_some() => quote! {
+                    #name::#con => {
+                        Err(msg("tags are not allowed for `flat` variants with no fields"))
```

Comment:
Why is this inside of `quote!` and where is `msg` defined?

Reply:
I think tag `flat` and `tag` are mutually exclusive on variant level, not just when there is only a unit constructor, because the tag would apply to the first field value and can therefore potentially collide with the `tag` attribute of the field. It is best to error in `attrs.rs` on the combination of `flat` and `tag` on variant level.

Reply:
> Why is this inside of `quote!` and where is `msg` defined?

I was mistaken with `encode::Error`.